### PR TITLE
[1LP][RFR] Improve email tests.

### PR DESCRIPTION
**Pull request by jaryn**
## Purpose or Intent
For checking CC and Bcc we need improvement of mock SMTP server we have.
- __Fixing__ GH-8820 
- __Adding__ coverage for BZ-1783511

Tests ran on with my jenkins job due to problems with connection to the PRT container:
https://cfmeqe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/downstream-511-tests-jhenner/19/

In above, the virtualcenter provision_approval tests had problems to pass. They seem to be flaky as seen here:
https://cfmeqe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/downstream-511-tests-jhenner/19/testReport/cfme.tests.cloud_infra_common/test_provisioning/
The failure seem not related:
```
>       wait_for(lambda: self.server_roles == roles, num_sec=timeout, delay=15)
E       wait_for.TimedOutError: Could not do 'lambda defined as `wait_for(lambda: self.server_roles == roles, num_sec=timeout, delay=15)`' at /var/ci/workspace/downstream-511-tests-jhenner/integration_tests/cfme/utils/appliance/__init__.py:1935 in time
```
Locally passed: cfme/tests/cloud_infra_common/test_provisioning.py::test_provision_approval[virtualcenter-6.7-edit] PASSED

{{py.test: -sv cfme/tests/cloud_infra_common/test_provisioning.py::test_provision_approval cfme/tests/configure/test_email.py::test_send_test_email --use-provider complete }}

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 06ea73e1643c7c7639552fbfd12f121a0c5ec0ff

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit b6812d861daa56c26ea7eb4d9ee00fdbd5d08d6a

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 0ce5b0bb07dee48307a08aa0e5eaff76a18ba9f1

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 901db74e04e20ea41ff976df0e32c4808c5b9f8b

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit dca6ecba4ece62be634bd4536a0950a6a696a425

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit e31845cdbef066308cbb137b8891d808958f72b2

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 6062a7e5dc11916be67d10fbb8ffb263e670fdea

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 6bde4e8118ff8f699a2b5119f43e18061a608aa3

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit baf3bcd145d8d9ab53eef16793e062596d93dab7

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 002f0045e1dd4f6c47a1033bafbe0763e5d3f297

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit e65bf95623f5eb159e5c52617e10ce1016c24434

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 7cf48f60182ca51a7e0c419e6ba58822ff18c01c

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 9e3d9f9715c6fc07496ea9923bcc855d668f679a

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit af772ca475037539411148c753a8168ba4c2a447

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/cloud_infra_common/test_provisioning.py [hunk 1]
```diff
@@ -17,11 +22,17 @@
 from cfme.utils import normalize_text
 from cfme.utils.appliance.implementations.ui import navigate_to
 from cfme.utils.blockers import BZ
+from cfme.utils.blockers import GH
 from cfme.utils.generators import random_vm_name
 from cfme.utils.log import logger
 from cfme.utils.providers import ProviderFilter
 from cfme.utils.update import update
 from cfme.utils.wait import wait_for
+logger.setLevel(logging.DEBUG)
```
Comment (mshriver) [inline cfme/tests/cloud_infra_common/test_provisioning.py:31]: Is this a statement you added for test debug, or do you intend to set logger to debug level when this module is read?

test_modules_importable, as well as pytest collection alone, will read this module - resulting in forced debug logging level for the remainder of the process.

Comment (jaryn) [reply cfme/tests/cloud_infra_common/test_provisioning.py:31]: I forgot to remove this. Thanks.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 9930aa9a6cbc7c6550631c15218e23b612c474f6

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 5278d32df75a369a80573418919589594931bf36

The *local* fixture **`email_addresses_configuration`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_provisioning.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Pull request merged**